### ISSUE TITLE
Snippet transformation does not work with nested snippets

### ISSUE THREAD

**Issue by johnknoop** [microsoft/vscode#112362]
VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.18363

I've written a snippet that transforms TypeScript interfaces to JSON key-values where the key is the name of the TS property and the value is the name of the type.

Example:

```
// input:
{
    myFirstProp: string;
    mySecondProp: number;
}
// output:
{
    myFirstProp: 'string',
    mySecondProp: 'number',
}
```

This is the snippet source:
```
"body": [
	"$1",
	"\n${1/([A-Za-z0-9]+): ([A-Za-z]+).*/$1: '$2',/gm}"
]
```

When the snippet is active and I'm typing the TS interface, I get the suggestion/autocomplete while typing. If I hit ESC before typing the semicolon that ends each line, then the snippet works great. But if I don't hit ESC, the snippet stops executing.

Steps to Reproduce:

I've recorded two gifs using a keylogger that lets you see which keys I'm hitting. Note that I'm on a Swedish keyboard, which the keylogger doesn't seem to support, so when I type `:` it will display it as `>` and when I type `;` it will display `<`.

Here's when I'm hitting escape before the semicolon:

![with_escape](https://user-images.githubusercontent.com/3394270/101987053-09120600-3c92-11eb-8677-11ddd9664925.gif)

Here's when I don't hit escaoe before the semicolon, which makes the snippet stop executing:

![without_escape](https://user-images.githubusercontent.com/3394270/101987076-2515a780-3c92-11eb-8e43-ca6f5b731e04.gif)

**Comment by jrieken** [comment]
Yeah, this is a bug that snippet transformations don't work on nested snippet invocation (which completions are)

**Comment by jrieken** [comment]
> Note that I'm on a Swedish keyboard, which the keylogger doesn't seem to support, so when I type : it will display it as > and when I type ; it will display <.

fyi - we have "F1 > Toggle Screencast Mode" which might be more true wrt what you are typing

**Comment by jrieken** [comment]
@johnknoop just to clarify: you have this set to false, right: `editor.suggest.snippetsPreventQuickSuggestions`

**Comment by johnknoop** [comment]
@jrieken Yes

**Comment by gotjoshua** [comment]
> nested snippet invocation

is there any progress on nested snippets and transformations finding harmony together?

I'd love to have a snippet that can use something like this:  
`${2:${TM_SELECTED_TEXT:${CLIPBOARD:arrayName}/(.)(.*)s/${1:/upcase}$2/}`. 
...nested alternatives to feed the transform
